src/videojs-hls.js

```diff
@@ -286,7 +286,18 @@ videojs.Hls.prototype.handleSourceOpen = function() {
  */
 videojs.Hls.prototype.play = function() {
   if (this.ended()) {
-    this.mediaIndex = 0;
+    // When the video ends and the 'ended' event fires, current time is set to 0. If the video is in this state, or the
+    // current time is equal to the duration (ie, play head is at the end position), reset the video to the beginning on
+    // a play event.
+    if (this.currentTime() === 0 || this.currentTime() === this.duration()) {
```

Comment:
If currentTime is zero, would this cause unnecessary re-buffering?

Are you sure that `currentTime === duration`? Could currentTime be 9.9999 and duration be 10, for instance? We support m3u8 annotations (`#ZEN-TOTAL-DURATION`) which might not be perfectly accurate.